modules/local/prokka.nf

```diff
@@ -0,0 +1,50 @@
+process PROKKA {
+
+    container "${ workflow.containerEngine in ['singularity', 'apptainer'] ?
+        'https://depot.galaxyproject.org/singularity/prokka:1.14.6--pl526_0' :
+        'biocontainers/prokka:1.14.6--pl526_0' }"
+
+    input:
+    tuple val(meta), path(assembly_file)
+
+    output:
+    tuple val(meta), path('prokka_out/contigs.gbk'), emit: prokka_gbk
+	tuple val(meta), path('prokka_out/contigs.gff'), emit: prokka_gff
+	tuple val(meta), path('prokka_out/contigs.faa'), emit: prokka_faa
+	tuple val(meta), path('prokka_out/contigs.fna'), emit: prokka_fna
+    path "versions.yml",                             emit: versions
+
+    script:
+    def is_compressed = assembly_file.getExtension() == "gz" ? true : false
+    def fasta_name = is_compressed ? assembly_file.getBaseName() : assembly_file
+
+    if(assembly_file.size() > 0)
```

Comment:
I think this kind of pattern it's not very good. This kind of control should be donde at the workflow level.

Reply:
I added this snippet in a few modules, namely prokka and proovframe for necessity. I took inspiration from Vangelis' diamond makedb. To understand, for example, would I need to add a snippet in `long_reads_assembly_qc.nf` to check for compressed files before the proovframe step?

Reply:
Yes, I think that pattern is preferred. IMO the logic should be in the workflows and not in the modules whenever possible